sql/src/main/java/io/crate/operation/collect/stats/StatsTablesService.java

```diff
@@ -0,0 +1,292 @@
+    volatile TimeValue lastOperationsLogExpiration;
+
+    static final JobContextLogSizeEstimator JOB_CONTEXT_LOG_ESTIMATOR = new JobContextLogSizeEstimator();
+    static final OperationContextLogSizeEstimator OPERATION_CONTEXT_LOG_SIZE_ESTIMATOR = new OperationContextLogSizeEstimator();
+
+    @Inject
+    public StatsTablesService(Settings settings,
+                              NodeSettingsService nodeSettingsService,
+                              ThreadPool threadPool,
+                              CrateCircuitBreakerService breakerService) {
+        this(settings, nodeSettingsService, threadPool.scheduler(), breakerService);
+    }
+
+    @VisibleForTesting
+    StatsTablesService(Settings settings,
+                       NodeSettingsService nodeSettingsService,
+                       ScheduledExecutorService scheduledExecutorService,
+                       CrateCircuitBreakerService breakerService) {
+        super(settings);
+        scheduler = scheduledExecutorService;
+        this.breakerService = breakerService;
+        nodeSettingsService.addListener(listener);
+
+        int jobsLogSize = CrateSettings.STATS_JOBS_LOG_SIZE.extract(settings);
+        TimeValue jobsLogExpiration = CrateSettings.STATS_JOBS_LOG_EXPIRATION.extractTimeValue(settings);
+        int operationsLogSize = CrateSettings.STATS_OPERATIONS_LOG_SIZE.extract(settings);
+        TimeValue operationsLogExpiration = CrateSettings.STATS_OPERATIONS_LOG_EXPIRATION.extractTimeValue(settings);
+
+        boolean isEnabled = CrateSettings.STATS_ENABLED.extract(settings);
+
+        initialIsEnabled = isEnabled;
+        initialJobsLogSize = jobsLogSize;
+        initialJobsLogExpiration = jobsLogExpiration;
+        initialOperationsLogSize = operationsLogSize;
+        initialOperationsLogExpiration = operationsLogExpiration;
+
+        lastIsEnabled = isEnabled;
+        lastJobsLogSize = jobsLogSize;
+        lastJobsLogExpiration = jobsLogExpiration;
+        lastOperationsLogSize = operationsLogSize;
+        lastOperationsLogExpiration = operationsLogExpiration;
+
+        statsTables = new StatsTables(this::isEnabled);
+        if (isEnabled()) {
+            setJobsLogSink(lastJobsLogSize, lastJobsLogExpiration);
+            setOperationsLogSink(lastOperationsLogSize, lastOperationsLogExpiration);
+        } else {
+            setJobsLogSink(0, TimeValue.timeValueSeconds(0L));
+            setOperationsLogSink(0, TimeValue.timeValueSeconds(0L));
+        }
+
+        this.breakerService.addBreakerChangeListener(CrateCircuitBreakerService.JOBS_LOG,
+            (CircuitBreaker breaker) -> setJobsLogSink(lastJobsLogSize, lastJobsLogExpiration)
+        );
+        this.breakerService.addBreakerChangeListener(CrateCircuitBreakerService.OPERATIONS_LOG,
+            (CircuitBreaker breaker) -> setOperationsLogSink(lastOperationsLogSize, lastOperationsLogExpiration)
+        );
+    }
+
+    private void setJobsLogSink(int size, TimeValue expiration) {
```

Comment:
@mfussenegger would it be necessary to set the new sink before closing the old one?

```java
    private void setJobsLogSink(int size, TimeValue expiration) {
        LogSink<JobContextLog> newSink = createSink(size, expiration, JOB_CONTEXT_LOG_ESTIMATOR,
            CrateCircuitBreakerService.JOBS_LOG);
        newSink.addAll(jobsLogSink);
        LogSink<JobContextLog> oldSink = jobsLogSink;
        jobsLogSink = newSink;
        statsTables.updateJobsLog(jobsLogSink);
        oldSink.close();
    }
```